genesyscloud/architect_schedules/resource_genesyscloud_architect_schedules.go

```diff
@@ -0,0 +1,262 @@
+	proxy := getArchitectSchedulesProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+	divisionID := d.Get("division_id").(string)
+	description := d.Get("description").(string)
+	start := d.Get("start").(string)
+	end := d.Get("end").(string)
+	rrule := d.Get("rrule").(string)
+
+	schedStart, err := time.Parse("2006-01-02T15:04:05.000000", start)
+	if err != nil {
+		return diag.Errorf("Failed to parse date %s: %s", start, err)
+	}
+
+	schedEnd, err := time.Parse("2006-01-02T15:04:05.000000", end)
+	if err != nil {
+		return diag.Errorf("Failed to parse date %s: %s", end, err)
+	}
+
+	schedule := platformclientv2.Schedule{
+		Name:  &name,
+		Start: &schedStart,
+		End:   &schedEnd,
+		Rrule: &rrule,
+	}
+
+	// Optional attributes
+	if description != "" {
+		schedule.Description = &description
+	}
+
+	if divisionID != "" {
+		schedule.Division = &platformclientv2.Writabledivision{Id: &divisionID}
+	}
+
+	log.Printf("Creating schedule %s", *schedule.Name)
+	scheduleResponse, proxyResponse, err := proxy.createArchitectSchedules(ctx, &schedule)
+	if err != nil {
+		msg := ""
+		if strings.Contains(fmt.Sprintf("%v", err), "routing:schedule:add") {
+			msg = "\nYou must have all divisions and future divisions selected in your OAuth client role"
+		}
+
+		return util.BuildAPIDiagnosticError("genesyscloud_archiect_schedules", fmt.Sprintf("Failed to create schedule %s | Error: %s MSG: %s", *scheduleResponse.Name, err, msg), proxyResponse)
+	}
+
+	d.SetId(*scheduleResponse.Id)
+
+	log.Printf("Created schedule %s %s", name, *scheduleResponse.Id)
+	return readArchitectSchedules(ctx, d, meta)
+}
+
+func readArchitectSchedules(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getArchitectSchedulesProxy(sdkConfig)
+
+	log.Printf("Reading schedule %s", d.Id())
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		scheduleResponse, proxyResponse, err := proxy.getArchitectSchedulesById(ctx, d.Id())
+		if err != nil {
+			if util.IsStatus404(proxyResponse) {
+				return retry.RetryableError(fmt.Errorf("failed to read schedule %s: %s", d.Id(), err))
+			}
+			return retry.NonRetryableError(fmt.Errorf("failed to read schedule %s: %s", d.Id(), err))
+		}
+
+		Start := new(string)
+		if scheduleResponse.Start != nil {
+			*Start = timeutil.Strftime(scheduleResponse.Start, "%Y-%m-%dT%H:%M:%S.%f")
+
+		} else {
+			Start = nil
+		}
+
+		End := new(string)
+		if scheduleResponse.End != nil {
+			*End = timeutil.Strftime(scheduleResponse.End, "%Y-%m-%dT%H:%M:%S.%f")
+
+		} else {
+			End = nil
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceArchitectSchedules(), constants.DefaultConsistencyChecks, resourceName)
+		d.Set("name", *scheduleResponse.Name)
+		d.Set("division_id", *scheduleResponse.Division.Id)
+		d.Set("description", nil)
+		if scheduleResponse.Description != nil {
```

Comment:
We try to use the `resourcedata` package when setting values like this, you can use `SetNillableValue` here instead of setting it directly, It avoids having to make the if statement yourself. You also don't need to set "description" to nil before doing this, `SetNillableValue` will do that if there is no description. Same for any other optional attributes. It wouldn't hurt to use for `SetNillableValue` for the required attributes as well, just incase the attribute gets changed to optional in the future it will stop any potential bugs

Reply:
Thanks for letting me know, 
Can I use GetNillableValue where we use Get instead?